src/components/Form/CreateProfileForm/DiscountData/EnrollToEyca.tsx

```diff
@@ -68,19 +60,51 @@ const EnrollToEyca = ({ profile, index, formValues, setFieldValue }: Props) => {
     }
   };
 
-  const openModal = (val: any) => {
-    setCheckboxValue(val);
+  const visibleOnEyca: boolean | undefined = hasIndex
+    ? formValues.discounts[index].visibleOnEyca
+    : formValues.visibleOnEyca;
+
+  const eycaLandingPageUrl: string | undefined = hasIndex
+    ? formValues.discounts[index].eycaLandingPageUrl
+    : formValues.eycaLandingPageUrl;
+
+  const openModal = () => {
     setIsModalOpen(true);
   };
 
   const closeModal = () => {
-    setFieldValue(
-      hasIndex ? `discounts[${index}].visibleOnEyca` : `visibleOnEyca`,
-      checkBoxValue
-    );
     setIsModalOpen(false);
   };
 
+  useEffect(() => {
+    // TODO sincronizza visible e landing
+
+    if (salesChannel.discountCodeType === DiscountCodeType.LandingPage) {
+      setFieldValue(
+        hasIndex ? `discounts[${index}].visibleOnEyca` : `visibleOnEyca`,
+        (eycaLandingPageUrl ?? "").trim() !== ""
+      );
+    } else {
+      setFieldValue(
+        hasIndex
+          ? `discounts[${index}].eycaLandingPageUrl`
+          : `eycaLandingPageUrl`,
+        ""
+      );
+    }
+  }, [
+    visibleOnEyca,
+    eycaLandingPageUrl,
+    salesChannel.discountCodeType,
+    setFieldValue,
+    hasIndex,
+    index
+  ]);
```

Comment:
What do you think to make this `useEffect` more readable with a solution like this:
```suggestion
   useEffect(() => {
    const visibleField = hasIndex
      ? `discounts[${index}].visibleOnEyca`
      : `visibleOnEyca`;

    const landingPageField = hasIndex
      ? `discounts[${index}].eycaLandingPageUrl`
      : `eycaLandingPageUrl`;

    if (salesChannel.discountCodeType === DiscountCodeType.LandingPage) {
      setFieldValue(visibleField, (eycaLandingPageUrl ?? "").trim() !== "");
    } else {
      setFieldValue(landingPageField, "");
    }
  }, [
    eycaLandingPageUrl,
    salesChannel.discountCodeType,
    setFieldValue,
    hasIndex,
    index
  ]);
```

Reply:
IMHO keeping field name inlined in setFieldValue() call site makes it easier to read.
I agree indentation is better in the suggestion, but it is a code indirection that would make it slower to read/modify/delete.
Let me know if you want this change anyway.